katdal/visdatav4.py

```diff
@@ -560,13 +560,20 @@ def _set_keep(self, time_keep=None, freq_keep=None, corrprod_keep=None,
         self._vis = DaskLazyIndexer(self._corrected.vis, stage1)
         self._weights = DaskLazyIndexer(self._corrected.weights, stage1)
         self._raw_flags = DaskLazyIndexer(self._corrected.flags, stage1)
+
+        # Create flags indexer based on current flag selection
         flag_transforms = []
         if ~self._flags_select != 0:
-            # Copy so that the lambda isn't affected by future changes
+            # Copy so that the closure isn't affected by future changes
             select = self._flags_select.copy()
-            flag_transforms.append(lambda flags: da.bitwise_and(select, flags))
-        flag_transforms.append(lambda flags: flags.view(np.bool_))
+            def bitwise_and(flags): return da.bitwise_and(select, flags)
+            flag_transforms.append(bitwise_and)
+        # Convert uint8 to bool by ORing all the bits
```

Comment:
It doesn't OR all the bits - they're still there. It's just that numpy treats any non-zero value as truthy.

Reply:
Would that survive the flag selection transformation?

Reply:
You would then only see the selected flags, but that's OK. You could just select `flags='all'` beforehand.

Reply:
```
In [30]: d.select(dumps=12, channels=slice(10,20), flags='static,cam')                             
In [38]: with ProgressBar(): 
    ...:     z = d.flags[0, :3, :5] 
    ...:                                                                                           
[########################################] | 100% Completed |  2min  5.8s

In [39]: z                                                                                         
Out[39]: 
array([[ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True]])
In [41]: import numpy as np                                                                        

In [42]: z.view(np.uint8)                                                                          
Out[42]: 
array([[2, 2, 2, 2, 2],
       [2, 2, 2, 2, 2],
       [2, 2, 2, 2, 2]], dtype=uint8)

In [43]: d.select(dumps=12, channels=slice(10,20), flags='all')                                    

In [44]: with ProgressBar(): 
    ...:     z = d.flags[0, :3, :5] 
    ...:                                                                                           
[########################################] | 100% Completed |  1min 15.7s

In [45]: z                                                                                         
Out[45]: 
array([[ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True]])

In [46]: z.view(np.uint8)                                                                          
Out[46]: 
array([[66, 66, 66, 66, 66],
       [66, 66, 66, 66, 66],
       [66, 66, 66, 66, 66]], dtype=uint8)
```

Reply:
The question is whether it is sufficient to tell end users about this trick. Or should we expose it like in `raw_flags`?

Reply:
I don't think that should be how end users access this - it's pretty fragile and would prevent us ever doing anything that collapsed the flags into a 0/1 bool.

Reply:
That's a con, agreed. A pro is that the trick is basically instant, while `raw_flags` needs another trip to the archive.

Reply:
That's fixable though - and anyway, it'll only make a difference if someone is loading both `flags` and `raw_flags` in the same get call, which seems like a fairly pointless use case since you could work directly off `raw_flags` and forgo `flags`.

Reply:
My use case is:
 - Use `flags` as before.
 - Ask: "why is that patch over there flagged?"
 - Invoke `raw_flags`.
 
I'm trying to shield the user from having to do bitwise ands and view-as-bools in the first place :-D